src/lib/components/events/event-carousel.svelte

```diff
@@ -9,12 +10,22 @@
 	let carouselButtonLeft: HTMLDivElement;
 	let carouselButtonRight: HTMLDivElement;
 	let isGrabbing = false;
+	let hasHorizontalScrollbar = false;
+	let leftButtonEnabled = false;
+	let rightButtonEnabled = false;
 
```

Comment:
Here, let's add this helper function to help us solve the bug with the arrow buttons:

```ts
let scrollTimeoutId = undefined;

const debounceTheScroll = (callback: () => void, wait: number) => {
  callback();
  if (scrollTimeoutId !== undefined) {
    clearTimeout(scrollTimeoutId);
  }
  scrollTimeoutId = setTimeout(callback, wait);
};
```

Reply:
Done.